hack/local-up-karmada-helm.sh

```diff
@@ -0,0 +1,239 @@
+NEED_CREATE_KIND_CLUSTER=${NEED_CREATE_KIND_CLUSTER:-"true"}
+NEED_CREATE_KIND_MEMBER_CLUSTER=${NEED_CREATE_KIND_MEMBER_CLUSTER:-"true"}
+CLUSTER_VERSION=${CLUSTER_VERSION:-"kindest/node:v1.27.3"}
+
+# IMAGE_FROM customize whether you need fetch images in advance, optional value are as follows:
+# pull-in-advance: use 'docker pull' to fetch needed images to local node in advance (in case of network trouble in automatically pulling images)
+# make: build karmada images from source code with latest tag and pull images (other than karmada) in advance ( in case of trying latest code)
+# empty or any other value: ignored, just pull image by k8s-runtime when needing
+IMAGE_FROM=${IMAGE_FROM:-"make"}
+
+# LOAD_IMAGE_IN_ADVANCE if you fetch images in advance and you are using KinD clusters, may be you'd like to load it to KinD in advance too
+# if not, please unset this option
+LOAD_IMAGE_IN_ADVANCE=${LOAD_IMAGE_IN_ADVANCE:-"true"}
+
+# KARMADA_HOST_NAME customize your cluster name and context name of karmada-host cluster, if you already has a cluster, replace it by your real name
+KARMADA_HOST_NAME=${KARMADA_HOST_NAME:-"karmada-host"}
+# KARMADA_HOST_KUBECONFIG customize your kubeconfig of karmada-host cluster, if you already has a cluster, replace it by your real kubeconfig path
+KARMADA_HOST_KUBECONFIG=${KARMADA_HOST_KUBECONFIG:-"${HOME}/.kube/karmada-host.config"}
+
+# PUSH_MODE_MEMBERS a map which customizing your push mode member clusters
+# the key of the map is the cluster name / context name of your member clusters
+# the value of the map is the corresponding kubeconfig path of the member cluster
+# if you already has member clusters, replace the key with real name and replace the value with real kubeconfig path
+# you can add any number push mode clusters as you expect, just append this map as following examples
+declare -A PUSH_MODE_MEMBERS
+PUSH_MODE_MEMBERS["member1"]="${HOME}/.kube/member1.config"
+PUSH_MODE_MEMBERS["member2"]="${HOME}/.kube/member2.config"
+
+# PULL_MODE_MEMBERS a map which customizing your pull mode member clusters
+# the key of the map is the cluster name / context name of your member clusters
+# the value of the map is the corresponding kubeconfig path of the member cluster
+# if you already has member clusters, replace the key with real name and replace the value with real kubeconfig path
+# you can add any number pull mode clusters as you expect, just append this map as following examples
+declare -A PULL_MODE_MEMBERS
+PULL_MODE_MEMBERS["member3"]="${HOME}/.kube/member3.config"
+
+# INSTALL_ESTIMATOR whether install karmada-scheduler-estimator
+INSTALL_ESTIMATOR=${INSTALL_ESTIMATOR:-"true"}
+
+echo "########## start installing karmada control plane ##########"
+set -ex
+
+# 1. create KinD cluster if you set NEED_CREATE_KIND_CLUSTER
+if ${NEED_CREATE_KIND_CLUSTER}; then
+  # 1.1 create karmada-host cluster
+  kind delete clusters karmada-host
```

Comment:
install the command `kind` first?

https://github.com/karmada-io/karmada/blob/8c832eab94160518a6b40e090c730e1ef54cf174/hack/cli-testing-environment.sh#L37-L45